client/scripts/views/components/component_kit/types.ts

```diff
@@ -0,0 +1,31 @@
+export enum ComponentType {
+  Button = 'Button',
+  GradientButton = 'GradientButton',
+}
+
+export enum ButtonType {
+  Primary = 'primary',
+  Secondary = 'secondary',
+}
+
+export type StyleProps = {
```

Comment:
Should be clarified as`ButtonStyleProps` as I assume there will be many different types of StyleProps defined, for each class of components